src/cartservice/Program.cs

```diff
@@ -163,21 +143,48 @@ static void Main(string[] args)
             }
         }
 
-        private static string ReadRedisAddress(string address)
+        /// <summary>
+        /// Reads parameter in the right order
+        /// </summary>
+        /// <param name="description">Parameter description</param>
+        /// <param name="commandLineValue">Value provided from the command line</param>
+        /// <param name="environmentVariableName">The name of environment variable where it could have been set</param>
+        /// <param name="environmentParser">The method that parses environment variable and returns typed parameter value</param>
+        /// <param name="defaultValue">Parameter's default value - in case other method failed to assign a value</param>
+        /// <typeparam name="T">The type of the parameter</typeparam>
+        /// <returns>Parameter value read from all the sources in the right order(priorities)</returns>
+        private static T ReadParameter<T>(
+            string description,
+            T commandLineValue, 
+            string environmentVariableName, 
+            Func<string, T> environmentParser, 
+            T defaultValue)
         {
-            if (!string.IsNullOrEmpty(address))
+            // Command line argument
+            if(!EqualityComparer<T>.Default.Equals(commandLineValue, default(T))) 
             {
-                return address;
+                return commandLineValue;
             }
 
-            Console.WriteLine($"Reading redis cache address from environment variable {REDIS_ADDRESS}");
-            string redis = Environment.GetEnvironmentVariable(REDIS_ADDRESS);
-            if (!string.IsNullOrEmpty(redis))
+            // Environment variable
+            Console.Write($"Reading {description} from environment variable {environmentVariableName}. ");
+            string envValue = Environment.GetEnvironmentVariable(environmentVariableName);
+            if (!string.IsNullOrEmpty(envValue))
             {
-                return redis;
+                try
+                {
+                    var envTyped = environmentParser(envValue);
+                    Console.WriteLine("Done!");
+                    return envTyped;
+                }
+                catch (Exception)
+                {
+                    // We assign the default value later on
```

Comment:
if we're gonna have so many logs for debugging CLI/env parsing, definitely log this exception message, otherwise there's no easy way of telling without modifying code or using debugger.

Reply:
There is already a log message that says which source assigns the value to the parameter, so if it's not environment variable, we'll discover it from the logs. No need to add anything in exception handling logic.